### ISSUE TITLE
never is confusing / not as readable in code

### ISSUE THREAD

**Issue by jimbo2150** [JordanRL/never-argument-type#1]
Having `never` as a function return type makes sense since the function is expected to always exit or otherwise never return.
For parameters, it seems confusing. `function foo(never $var)` seems like it should never have a `$var` parameter.

`never` having 2 different definitions is confusing. It meaning no further execution will occur for a function, and meaning a parameter requires a type is confusing. 

> As a side note, to me, it seems like it would be along the lines violating named parameters:
> ```
> foo(
>    reqVal: 1,
>    var: 3
> );
> ```
> and that it would throw an error because a parameter named $var was being passed in.

The TypeScript article linked in the RFC indicates when to use the base types:

>- Use `never` in positions where there will not or should not be a value.
>- Use `unknown` where there will be a value, but it might have any type.

Seems like `unknown` or possibly `typed` would be a better, more understandable fit that a parameter must have a type. Such a type could potentially just be an alias or extension of the `never` base type, but would be more readable for parameters.

- `function foo(unknown $var)` or `function(typed $var)` seems more understandable in code that the `$var` parameter would need to have a type defined.

**Comment by jpresutti** [comment]
I agree wholeheartedly. I'm not fond of this feature in the first place as it literally makes using interfaces correctly impossible, but calling it "never" is insane to me.

**Comment by khallmark** [comment]
Why should there be a language feature to fix an architecture problem?
